### PR TITLE
Bugfix #9860 [v98] Sentry cloud upgrade integration

### PR DESCRIPTION
**Pull request by yoanarios**
Issue #9860

- Upgrade to version 7.9.0 from 4.4.3 and fix breaking changes
- Create SentryProtocol for RatingPromptTest
- There is no need for 2 different DSN because we use the environment option to differentiate Nightly from release
- Updated bitrise to upload dSYM

### CODE REVIEW

File: ClientTests/RatingPromptManagerTests.swift [hunk 1]
```diff
@@ -16,6 +15,7 @@ class RatingPromptManagerTests: XCTestCase {
     var promptManager: RatingPromptManager!
     var mockProfile: MockProfile!
     var createdGuids: [String] = []
+    var sentry: CrashingMockSentryClient!
```
Comment (nbhasin2) [inline ClientTests/RatingPromptManagerTests.swift:18]: qq: is `CrashingMockSentryClient` a mock sentry client for testing? @yoanarios

Comment (yoanarios) [reply ClientTests/RatingPromptManagerTests.swift:18]: yes @nbhasin2, it's only used in RatingPromptTest for the logic to show rating prompt

File: Shared/SentryIntegration.swift [hunk 2]
```diff
@@ -54,44 +53,44 @@ public class Sentry {
             return
         }
 
-        var sentryDSNKey = SentryDSNKey
+        var environment = "Production"
         if AppInfo.appVersion == AppConstants.NIGHTLY_APP_VERSION, AppConstants.BuildChannel == .beta {
             // Setup sentry for Nightly Firefox Beta
-            sentryDSNKey = SentryNightlyDSNKey
+            environment = "Nightly"
```
Comment (nbhasin2) [inline Shared/SentryIntegration.swift:59]: nit: would be nice to make them enum or static vars so we can easily change the env

File: bitrise.yml [hunk 3]
```diff
@@ -900,6 +908,14 @@ workflows:
...
         - itunescon_user: "$APPLE_ACCOUNT_ID"
         title: 'Nightly FirefoxBeta: Xcode Archive & Export for iOS'
+    - script@1:
+        title: Upload Nightly Symbols
+        inputs:
+        - content: |-
+            #!/usr/bin/env bash
+            set -x
+            ThirdParty/sentry-cli --auth-token "$SENTRY_AUTH_TOKEN" upload-dif \
+              --org mozilla --project firefox-ios "$BITRISE_DSYM_DIR_PATH"
```
Comment (nbhasin2) [inline bitrise.yml:918]: @yoanarios we should add this script to `SPM_Deploy_Beta_Only` as well

**Review by nbhasin2 — APPROVED** [review]

File: bitrise.yml (hunk 3 above)
Comment (nbhasin2) [reply bitrise.yml:918]: https://github.com/mozilla-mobile/firefox-ios/pull/10105/files/b5373c272db9aab0bc9feec4f7d1617b7f754b07#diff-61d1ee831cb1df242161a7d23171914a1d5d628c7659b52db957d80a9394bd68R742

Comment (yoanarios) [reply bitrise.yml]: done, thank you

**Pull request merged**